### PR TITLE
chore(ocsp): hide ocsp from schema for now

### PR DESCRIPTION
**Pull request by thalesmg**
We're going to release it in a later version.  So, for now, we'll just hide it.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [~] Added tests for the changes
- [~] Changed lines covered in coverage report
- [~] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` and `.zh.md` files
- [~] For internal contributor: there is a jira ticket to track this change
- [~] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [X] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [~] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [~] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by zmstone — DISMISSED** [review]

**Comment by zmstone** [comment]
please confirm check if the new args show up in the dumped schema-*.json

**Comment by thalesmg** [comment]
> please confirm check if the new args show up in the dumped schema-*.json

Indeed, putting `hidden` at that point seems to have no effect.  :see_no_evil: 

I tried marking each sub-field as `hidden`, but the struct still appears in `schema-en.json`:

```json
     {
         "desc": "Per listener OCSP Stapling configuration.",
         "fields": [],
         "full_name": "broker:ocsp",
         "paths": [
             "gateway.coap.listeners.dtls.$name.dtls_options.ocsp",
             "gateway.exproto.listeners.dtls.$name.dtls_options.ocsp",
             "gateway.exproto.listeners.ssl.$name.ssl_options.ocsp",
             "gateway.lwm2m.listeners.dtls.$name.dtls_options.ocsp",
             "gateway.mqttsn.listeners.dtls.$name.dtls_options.ocsp",
             "gateway.stomp.listeners.ssl.$name.ssl_options.ocsp",
             "listeners.quic.$name.ssl_options.ocsp",
             "listeners.ssl.$name.ssl_options.ocsp"
         ],
         "tags": [
             "EMQX"
         ]
     },
```

Maybe hiding that would require a patch in `hocon_json` :thinking:

**Comment by thalesmg** [comment]
Trying to fix it here: https://github.com/emqx/hocon/pull/237

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -2066,6 +2066,10 @@ desc("ocsp") ->
 desc(_) ->
     undefined.
 
+%% TODO: remove after e5.0.2
+hidden("ocsp") -> true;
```
Comment (thalesmg) [inline apps/emqx/src/emqx_schema.erl:2070]: With this meta and [this patch](https://github.com/emqx/hocon/pull/237), the `ocsp` struct is removed from the generated JSON.

**Comment by zmstone** [comment]
We should not add OCSP to quic listeners as we do not have control over the SSL handshakes there.

**Comment by qzhuyan** [comment]
> We should not add OCSP to quic listeners as we do not have control over the SSL handshakes there.

soon, we will!

File: apps/emqx/src/emqx_schema.erl [hunk 2]
```diff
@@ -1285,6 +1285,8 @@ fields("listener_wss_opts") ->
     );
 fields("listener_quic_ssl_opts") ->
     %% Mark unsupported TLS options deprecated.
+    Schema0 = server_ssl_opts_schema(#{}, false),
+    Schema1 = lists:keydelete("ocsp", 1, Schema0),
```
Comment (lafirest) [inline apps/emqx/src/emqx_schema.erl:1289]: maybe we could combine the `lists:keydelete` and `lists:map` into `lists:foldX`

**Review by lafirest — APPROVED** [review]

**Review by zmstone — DISMISSED** [review]

**Pull request merged**